src/client/utils/collections.js

```diff
@@ -0,0 +1,78 @@
+import { useQuery } from 'react-query';
+import collectionPlaceholderPng from '../resources/icon.png';
+import { getCollection, getCollections } from '../../api/collection';
+import {
+  CACHE_TIME_MILLISECONDS,
+  COLLECTIONS,
+  STALE_TIME_MILLISECONDS,
+} from '../config/constants';
+
+const collectionQueryConfig = {
+  staleTime: STALE_TIME_MILLISECONDS, // time until data in cache considered stale if cache not invalidated
+  cacheTime: CACHE_TIME_MILLISECONDS, // time before cache labeled as inactive to be garbage collected
+};
+
+// fallback collection
+const loadingCollection = {
+  name: 'Loading...',
+  description: 'loading...',
+  loadingCollection: true,
+  comments: [],
+  subitems: [],
+  voteScore: 0,
+  views: 0,
+  image: {
+    backgroundUrl: collectionPlaceholderPng,
+    pictureId: collectionPlaceholderPng,
+    thumbnailUrl: collectionPlaceholderPng,
+  },
+  author: {
+    id: '1',
+    name: 'Loading...',
+    image: {
+      pictureId: collectionPlaceholderPng,
+      thumbnailUrl: collectionPlaceholderPng,
+      gravatarUrl: collectionPlaceholderPng,
+    },
+  },
+  members: [
+    {
+      id: '1',
+      name: 'Loading ...',
+      type: 'owner',
+      image: {
+        pictureId: collectionPlaceholderPng,
+        thumbnailUrl: collectionPlaceholderPng,
+        gravatarUrl: collectionPlaceholderPng,
+      },
+    },
+  ],
+};
+
+const loadingCollections = Array.from({ length: 10 }, (v, index) => ({
+  id: `loading-collection-${index}`,
+  ...loadingCollection,
+}));
+
+function useCollection(collectionId) {
+  const { data, isLoading } = useQuery({
+    queryKey: [COLLECTIONS, collectionId], // cache key for collection detailes
+    queryFn: () => getCollection(collectionId).then((datas) => datas),
+    ...collectionQueryConfig,
+  });
+  return { collection: data ?? loadingCollection, isLoading };
```

Comment:
I checked a bit the docs, and would the `placeholderData` works? https://react-query.tanstack.com/guides/placeholder-query-data There seems to have a `isPlaceholderData` as well https://react-query.tanstack.com/reference/useQuery

Reply:
This could more easily allow to return `useQuery` directly so we can have access to all its data depending on our needs (`isFetchedAfterMount`, `isFetching`, `isIdle`, ...)

Reply:
I agree, but I have to experiment more with it first locally. Not sure if this should be used for our same use case. Anyways this can expand depending on our needs as you said.